tests/TestCase/TestSuite/TestCaseTest.php

```diff
@@ -146,6 +146,48 @@ public function testWithErrorReportingWithException(): void
         }
     }
 
+    /**
+     * test deprecated
+     */
+    public function testDeprecated(): void
+    {
+        $this->deprecated(function () {
+            trigger_error('deprecation message', E_USER_DEPRECATED);
+        });
+    }
+
+    /**
+     * test deprecated with assert after trigger warning
+     */
+    public function testDeprecatedWithAssertAfterTriggerWarning(): void
+    {
+        try {
+            $this->deprecated(function () {
+                trigger_error('deprecation message', E_USER_DEPRECATED);
+                $this->assertTrue(false, 'A random message');
```

Comment:
this testCase tries to show `deprecated()` behave unlike the phpunit's `$this->expectWarning();`